### ISSUE TITLE
[Android] [WASM] Wrapped `TextBlock` takes all available width

### ISSUE THREAD

**Issue by davidjohnoliver** [unoplatform/uno#5773]
## How to reproduce it (as minimally and precisely as possible)

Use the following Xaml:

```xml

	<Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
		  MaxWidth="350"
		  BorderBrush="Black"
		  BorderThickness="1">
		<StackPanel HorizontalAlignment="Center"
					Background="LightPink"
					VerticalAlignment="Top">
			<TextBlock Text="▮▮▮▮▮▮▮ ▮▮▮▮▮▮▮ ▮▮▮▮▮▮▮ ▮▮▮▮▮▮▮"
					   TextWrapping="WrapWholeWords"
					   TextAlignment="Center"
					   FontSize="30" />
		</StackPanel>
	</Grid>
```

## Current behavior

TextBlock uses all available width on some platforms.

### Android:

![image](https://user-images.githubusercontent.com/8270914/115218873-ff336000-a0d4-11eb-9552-5b84d3e55be5.png)

### WASM:

![image](https://user-images.githubusercontent.com/8270914/115219076-2a1db400-a0d5-11eb-8042-237ccbd24272.png)

## Expected behavior

TextBlock should only use as much width as needed to display wrapped text.

### UWP:

![image](https://user-images.githubusercontent.com/8270914/115219423-84b71000-a0d5-11eb-95f4-8c59c1953d7d.png)

### iOS:

![image](https://user-images.githubusercontent.com/8270914/115219657-baf48f80-a0d5-11eb-96d4-a68d902dabfa.png)

## Environment

Nuget Package:

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s): 3.7.0-dev.251

Affected platform(s):

- [ ] iOS
- [x] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by davidjohnoliver** [comment]
Possibly related to https://github.com/unoplatform/uno/issues/4455?

**Comment by jeromelaban** [comment]
In which context does this happen? It may also be related to Grid ?

**Comment by davidjohnoliver** [comment]
> In which context does this happen? It may also be related to Grid ?

I don't think so, I just checked and it's also bugged if you replace the `Grid` with a `Border`. The XAML above is the content of MainPage in a blank app.

**Comment by MartinZikmund** [comment]
No longer relevant, works correctly for Skia renderer targets
